src/shared_state.rs

```diff
@@ -370,3 +416,11 @@ fn empty_or_none(field: &Option<&String>) -> bool {
         None => true,
     }
 }
+
+#[inline]
+fn empty_or_none_owned(field: &Option<String>) -> bool {
```

Comment:
Would be better if we didn't need this function. The fact that `Connect<'_>` has a lifetime and is borrowing temporary variables makes the code a bit more convoluted that it needs to be, I think.

What if instead of having fields of type `Option<&String>` we simply had `Option<String>` (talking about `user`, `pass`, `auth_token`, etc)? It's not like saving a few allocations in the options struct saves us any tangible performance...

Reply:
Makes sense.